controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -0,0 +1,136 @@
+	ret := []*cephv1.CephRBDMirror{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateNameForCephRbdMirror(initData),
+				Namespace: initData.Namespace,
+			},
+			Spec: cephv1.RBDMirroringSpec{
+				Count:     1,
+				Resources: defaults.GetDaemonResources("rbd-mirror", initData.Spec.Resources),
+			},
+		},
+	}
+	for _, obj := range ret {
+		err := controllerutil.SetControllerReference(initData, obj, r.Scheme)
+		if err != nil {
+			r.Log.Error(err, "Unable to set controller reference for CephRBDMirror.", "CephRBDMirror", klog.KRef(obj.Namespace, obj.Name))
+			return nil, err
+		}
+	}
+	return ret, nil
+}
+
+// ensureCreated ensures that cephRbdMirror resources exist in the desired state.
+func (obj *ocsCephRbdMirrors) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	reconcileStrategy := ReconcileStrategy(instance.Spec.ManagedResources.CephRbdMirrors.ReconcileStrategy)
+	if reconcileStrategy == ReconcileStrategyIgnore {
+		return nil
+	}
+
+	cephRbdMirrors, err := r.newCephRbdMirrorInstances(instance)
+	if err != nil {
+		return err
+	}
+
+	if !instance.Spec.Mirroring.Enabled {
+		return r.deleteCephRbdMirrorInstance(cephRbdMirrors)
+	}
+
+	for _, cephRbdMirror := range cephRbdMirrors {
+		existing, err := r.fetchCephRbdMirrorInstance(cephRbdMirror)
+
+		if err != nil {
+			if errors.IsNotFound(err) {
+				r.Log.Info("Creating CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+				err = r.Client.Create(context.TODO(), cephRbdMirror)
+				if err != nil {
+					r.Log.Error(err, "Failed to create CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+					return err
+				}
+				continue
+			}
+			r.Log.Error(err, "Unable to create CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+			return err
+		}
+
+		if reconcileStrategy == ReconcileStrategyInit {
+			return nil
+		}
+
+		if existing.DeletionTimestamp != nil {
+			r.Log.Info("Unable to restore CephRbdMirror, It is marked for deletion.", "CephRbdMirror", klog.KRef(existing.Namespace, existing.Name))
+			return fmt.Errorf("failed to restore initialization object %s, It is marked for deletion", existing.Name)
+		}
+
+		r.Log.Info("Restoring original CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+		existing.ObjectMeta.OwnerReferences = cephRbdMirror.ObjectMeta.OwnerReferences
+		cephRbdMirror.ObjectMeta = existing.ObjectMeta
+		err = r.Client.Update(context.TODO(), cephRbdMirror)
+		if err != nil {
+			r.Log.Error(err, "Failed to update CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+			return err
+		}
+	}
+	return nil
+}
+
+// ensureDeleted deletes the CephRbdMirrors owned by the StorageCluster
+func (obj *ocsCephRbdMirrors) ensureDeleted(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) error {
```

Comment:
should not we check strategy in the deletion part as well

Reply:
Yes, we should.